test/features/support/ui_oauth_details.js

```diff
@@ -0,0 +1,199 @@
+const pactum = require("pactum");
+const { Given, When, Then, Before, After } = require("@cucumber/cucumber");
+const {
+  localhost,
+  requestTime,
+  validScope,
+  validResponseType,
+  validClientId,
+  validRedirectUri,
+} = require("./helpers/helpers");
+
+let specUIOAuthDetails;
+let invalidScope;
+let invalidResponseType;
+let invalidClientId;
+let invalidRedirectUri;
+
+const baseUrl = `${localhost}authorization/oauth-details`;
+
+const requestFunction = (scope, responseType, clientId, redirectUri) =>
+  specUIOAuthDetails.post(baseUrl).withBody({
+    requestTime: requestTime,
+    request: {
+      scope: scope,
+      responseType: responseType,
+      clientId: clientId,
+      redirectUri: redirectUri,
+      display: "popup",
+      prompt: "login",
+      acrValues: "mosip:idp:acr:static-code mosip:idp:acr:generated-code",
+      claims: {
+        userinfo: {
+          given_name: {
+            essential: true,
+          },
+          phone: null,
+          email: {
+            essential: true,
+          },
+        },
+        id_token: {},
+      },
+    },
+  });
+
+const errorResultFunction = async (errorCode) => {
+  await specUIOAuthDetails.toss();
+  specUIOAuthDetails.response().should.have.status(400);
+  specUIOAuthDetails.response().should.have.jsonLike({
+    responseTime: "2022-09-22T08:03:45.287Z",
+    response: null,
+    errors: [
+      {
+        errorCode: errorCode,
+      },
+    ],
+  });
+};
+
+Before(() => {
+  specUIOAuthDetails = pactum.spec();
+});
+
+// Scenario: Successfully validates the provided request parameters and resolves the required authentication factors
+Given(
+  "The user wants to validate the provided request parameters and resolve the required authentication factors",
+  () => "Every provided query parameters are valid"
+);
+
+When("The user triggers an action with the required parameters", () =>
+  requestFunction(
+    validScope,
+    validResponseType,
+    validClientId,
+    validRedirectUri
+  )
+);
+
+Then(
+  "The user successfully validates the provided request parameters and resolve the required authentication factors",
+  async () => {
+    await specUIOAuthDetails.toss(),
+      specUIOAuthDetails.response().should.have.status(200),
+      specUIOAuthDetails.response().should.have.jsonLike({
+        responseTime: "2022-09-22T08:03:45.287Z",
+        response: {
+          transactionId: "string",
+          authFactors: [
+            {
+              type: "PIN",
+            },
+          ],
+          essentialClaims: ["all claims"],
+        },
+        errors: [],
```

Comment:
![image](https://user-images.githubusercontent.com/119664031/214903871-c3ee66fb-7056-45b0-923a-f70d082afaa4.png)


seems like ```null``` should be the default value, also to change in mocks
```[] -> null```